packages/components/src/segmented-control/radio-context.ts

```diff
@@ -0,0 +1,13 @@
+/**
+ * WordPress dependencies
+ */
+import { createContext } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import type { SegmentedControlRadioState } from './types';
+
+const RadioContext = createContext( {} as SegmentedControlRadioState );
+
+export default RadioContext;
```

Comment:
Nit: we usually name the context after the component that uses it, like this might be better named `SegmentedControlContext`.

We also usually add a `use...` function:

```suggestion
const SegmentedControlContext = createContext( {} as SegmentedControlRadioState );

export const useSegmentedControlContext = () => useContext(SegmentedControlContext);

export default SegmentedControlContext;
```

We may also want to rename SegmentedControlRadioState to SegmentedControlContext.